### DISCUSSION TITLE
v1.5.0-rc.0

### DISCUSSION THREAD

**Discussion by ruslandoga** [plausible/analytics#2324] [Anything Else]
👋

The highlights of this release are:

- City and region level geolocation (disabled by default, please see the comments below for the guide)
- Google Analytics integration: https://plausible.io/docs/google-analytics-import (WIP setup guide: https://github.com/plausible/docs/pull/278)
- CSV data export: https://plausible.io/docs/export-stats
- ARM64 support

### Upgrading Plausible Analytics to v1.5

In your `docker-compose.yml` update the image used for `plausible` to tag `1.5.0-rc.0`:

```diff
plausible:
- image: plausible/analytics:v1.4
+ image: plausible/analytics:1.5.0-rc.0
````

#### Restart the container

```
$ docker compose stop plausible
$ docker compose rm plausible
$ docker compose up -d
```

This will boot up the new version of the app.

### Upgrading ClickHouse to 22.6 (Optional, but recommended)

In your `docker-compose.yml`, update the image used for `plausible_events_db` to the newest stable ClickHouse version:

```diff
plausible_events_db:
- image: yandex/clickhouse-server:21.3.2.5
+ image: clickhouse/clickhouse-server:22.6-alpine
```

#### Restart the container

```
$ docker compose stop plausible plausible_events_db
$ docker compose rm plausible_events_db
$ docker compose up -d
```

This will boot up the new version of ClickHouse.

Related PR: https://github.com/plausible/hosting/pull/45

### Upgrading PostgreSQL to 14 (Optional, but recommended)

Upgrading PostgreSQL is a bit trickier, [here's the upgrade guide.](https://github.com/plausible/hosting/blob/master/upgrade/postgres.md)

## Changelog

Following changes have been made since v1.4:

### Added
- The ability to escape `|` characters with `\` in Stats API filter values
- An upper bound of 1000 to the `limit` parameter in Stats API
- The `exclusions` script extension now also takes a `data-include` attribute tag
- A `file-downloads` script extension for automatically tracking file downloads as custom events
- Integration with [Matomo's referrer spam list](https://github.com/matomo-org/referrer-spam-list/blob/master/spammers.txt) to block known spammers
- API route `PUT /api/v1/sites/goals` with form params `site_id`, `event_name` and/or `page_path`, and `goal_type` with supported types `event` and `page`
- API route `DELETE /api/v1/sites/goals/:goal_id` with form params `site_id`
- The public breakdown endpoint can be queried with the "events" metric
- Data exported via the download button will contain CSV data for all visible graps in a zip file.
- Region and city-level geolocation plausible/analytics#1449
- The `u` option can now be used in the `manual` extension to specify a URL when triggering events.
- Delete a site and all related data through the Sites API
- Subscribed users can see their Paddle invoices from the last 12 months under the user settings
- Allow custom styles to be passed to embedded iframe plausible/analytics#1522
- New UTM Tags `utm_content` and `utm_term` plausible/analytics#515
- If a session was started without a screen_size it is updated if an event with screen_size occurs
- Added `LISTEN_IP` configuration parameter plausible/analytics#1189
- The breakdown endpoint with the property query `property=event:goal` returns custom goal properties (within `props`)
- Added IPv6 Ecto support (via the environment-variable `ECTO_IPV6`)
- New filter type: `contains`, available for `page`, `entry_page`, `exit_page`
- Add filter for custom property
- Add ability to import historical data from GA: plausible/analytics#1753
- API route `GET /api/v1/sites/:site_id`
- Hovering on top of list items will now show a [tooltip with the exact number instead of a shortened version](https://github.com/plausible/analytics/discussions/1968)
- Filter goals in realtime filter by clicking goal name
- The time format (12 hour or 24 hour) for graph timelines is now presented based on the browser's defined language
- Choice of metric for main-graph both in UI and API (visitors, pageviews, bounce_rate, visit_duration) plausible/analytics#1364
- New width=manual mode for embedded dashboards plausible/analytics#2148
- Add more timezone options
- Add new strategy to recommend timezone when creating a new site
- Alert outgrown enterprise users of their usage plausible/analytics#2197
- Manually lock and unlock enterprise users plausible/analytics#2197
- ARM64 support for docker images plausible/analytics#2103
- Add support for international domain names (IDNs) plausible/analytics#2034

### Fixed
- Plausible script does not prevent default if it's been prevented by an external script [plausible/analytics#1941](https://github.com/plausible/analytics/issues/1941)
- Hash part of the URL can now be used when excluding pages with `script.exclusions.hash.js`.
- UI fix where multi-line text in pills would not be underlined properly on small screens.
- UI fix to align footer columns
- Guests can now use the favicon to toggle additional info about the site bing viewed (such as in public embeds).
- Fix SecurityError in tracking script when user has blocked all local storage
- Prevent dashboard graph from being selected when long pressing on the graph in a mobile browser
- The exported `pages.csv` file now includes pageviews again [plausible/analytics#1878](https://github.com/plausible/analytics/issues/1878)
- Fix a bug where city, region and country filters were filtering stats but not the location list
- Fix a bug where regions were not being saved
- Timezone offset labels now update with time changes
- Render 404 if shared link auth cannot be verified [plausible/analytics#2225](https://github.com/plausible/analytics/pull/2225)
- Restore compatibility with older format of shared links [plausible/analytics#2225](https://github.com/plausible/analytics/pull/2225)
- Fix 'All time' period for sites with no recorded stats [plausible/analytics#2277](https://github.com/plausible/analytics/pull/2277)
- Ensure settings page can be rendered after a form error [plausible/analytics#2278](https://github.com/plausible/analytics/pull/2278)

### Changed
- `script.file-downloads.outbound-links.js` only sends an outbound link event when an outbound download link is clicked
- Plausible script now uses callback navigation (instead of waiting for 150ms every time) when sending custom events
- Cache the tracking script for 24 hours
- Move `entry_page` and `exit_page` to be part of the `Page` filter group
- Paginate /api/sites results and add a `View all` link to the site-switcher dropdown in the dashboard.
- Remove the `+ Add Site` link to the site-switcher dropdown in the dashboard.
- `DISABLE_REGISTRATIONS` configuration parameter can now accept `invite_only` to allow invited users to register an account while keeping regular registrations disabled plausible/analytics#1841
- New and improved Session tracking module for higher throughput and lower latency. [PR#1934](https://github.com/plausible/analytics#1934)
- Do not display ZZ country code in countries report [PR#1934](https://github.com/plausible/analytics#2223)
- Add fallback icon for when DDG favicon cannot be fetched [PR#2279](https://github.com/plausible/analytics#2279)

<hr /><em>This discussion was created from the release <a href='https://github.com/plausible/analytics/releases/tag/v1.5.0-rc.0'>v1.5.0-rc.0</a>.</em>

**Comment by rphlo** [comment]
Can you describe how to enable city level geolocation? And if it is disabled wouldn't it be possible to hide the buttons "Regions" and "Cities" on the Location widget?

**Comment by ruslandoga** [comment]
@rphlo sure! Here's a quick guide.

⚠️ Note that this city level geolocation feature requires extra **~1GB of RAM** to work. If you don't have enough RAM available, you might experience slow container startups and restarts.

⚠️ `dbip-city-lite` might not include all the necessary info for this feature, consider using maxmind, see replies below for details. 

---

City level geolocation can be enabled by mounting a volume with the downloaded [`geonames.csv`](https://github.com/plausible/location#cities) and [`dbip-city-lite-2022-10.mmdb`](https://db-ip.com/db/download/ip-to-city-lite) and setting `GEONAMES_SOURCE_FILE` and `IP_GEOLOCATION_DB` to point to them:

```console
$ # custom csv can be used, see https://github.com/plausible/location#cities for more info
$ curl -O https://s3.eu-central-1.wasabisys.com/plausible-application/geonames.csv
```
```console
$ # https://db-ip.com/db/download/ip-to-city-lite, maxmind can also be used but it requires authorization
$ curl -O https://download.db-ip.com/free/dbip-city-lite-2022-10.mmdb.gz
$ gunzip dbip-city-lite-2022-10.mmdb.gz
```

`docker-compose.yml`
```diff
  plausible:
    image: plausible/analytics:1.5.0-rc.0
+   volumes:
+     - ./geonames.csv:/etc/app/geonames.csv:ro
+     - ./dbip-city-lite-2022-10.mmdb:/etc/app/dbip-city-lite-2022-10.mmdb:ro
```

`plausible-conf.env`
```diff
ADMIN_USER_EMAIL=replace-me
ADMIN_USER_NAME=replace-me
ADMIN_USER_PWD=replace-me
BASE_URL=replace-me
SECRET_KEY_BASE=replace-me
+ GEONAMES_SOURCE_FILE=/etc/app/geonames.csv
+ IP_GEOLOCATION_DB=/etc/app/dbip-city-lite-2022-10.mmdb
```

**Reply by madanyang** [reply]
How does one enable city names using Maxmind which is more accurate then the db-ip

**Reply by ruslandoga** [reply]
@madanyang same way, by pointing to locally mounted maxmind's `mmdb` with `IP_GEOLOCATION_DB`.

**Reply by timscha** [reply]
I did it exactly as written but Plausible is crashing after these changes:

```
plausible-plausible-1            | Crash dump is being written to: erl_crash.dump...Loading plausible..
plausible-plausible-1            | Starting dependencies..
plausible-plausible-1            | Starting repos..
plausible-plausible-1            | create Plausible.Repo database if it doesn't exist
plausible-plausible-1            | create Plausible.ClickhouseRepo database if it doesn't exist
plausible-plausible-1            | Creation of Db successful!
plausible-plausible-1            | Loading plausible..
plausible-plausible-1            | Starting dependencies..
plausible-plausible-1            | Starting repos..
plausible-plausible-1            | Running migrations for Elixir.Plausible.Repo
plausible-plausible-1            | Running migrations for Elixir.Plausible.ClickhouseRepo
plausible-plausible-1            | Migrations successful!
plausible-plausible-1            | Loading plausible..
plausible-plausible-1            | Starting dependencies..
plausible-plausible-1            | Starting repos..
plausible-plausible-1            | Admin user already exists. I won't override, bailing
plausible-plausible-1            | 10:49:15.157 [warning] OTLP exporter failed to initialize with exception :throw:{:application_either_not_started_or_not_ready, :tls_certificate_check}
plausible-plausible-1            | {"Kernel pid terminated",application_controller,"{application_start_failure,plausible,{bad_return,{{'Elixir.Plausible.Application',start,[normal,[]]},{'EXIT',{timeout,{'Elixir.Task',await,[#{'__struct__' => 'Elixir.Task',owner => <0.2926.0>,pid => <0.2930.0>,ref => #Ref<0.451273697.225771521.164627>},30000]}}}}}}"}
plausible-plausible-1            | Kernel pid terminated (application_controller) ({application_start_failure,plausible,{bad_return,{{'Elixir.Plausible.Application',start,[normal,[]]},{'EXIT',{timeout,{'Elixir.Task',await,[#{'__struct__' => 'Elixir.Task',owner => <0.2926.0>,pid => <0.2930.0>,ref => #Ref<0.451273697.225771521.164627>},30000]}}}}}})
```

This repeats all the time. If I remove the changes in docker-compose and env file Plausible is working again. I downloaded the files, paths are correct too.

**Reply by ruslandoga** [reply]
@timscha 👋

How much memory is allocated to the plausible container? I had this error -- [`Location.load_all`](https://github.com/plausible/analytics/blob/ae4ae5d0a0b49ee0fbd5146cdd169455c6924f3d/lib/plausible/application.ex#L34) timeout -- when my container didn't have enough resources available, it was running unrestricted on a `t3.micro` instance.

**Reply by timscha** [reply]
@ruslandoga  Hi! :) I don't limit the memory at all (as I know. 😆). The whole server has 4 GB left of memory.

**Reply by ruslandoga** [reply]
For reference, [here's the part where timeout (30s) occurs.](https://github.com/plausible/location/blob/ca240782e98adcbb7470ef37c5f5ebb3619528b8/lib/location.ex#L9-L21) I'll go and try debugging it :)

**Reply by timscha** [reply]
@ruslandoga  If I can help somehow to provide further details or help debugging please let me know. :)

**Reply by ruslandoga** [reply]
@timscha sorry for the long wait. I've tried running the docker-compose setup on `t3.medium` (2 vcpu, 4GB ram), `t3.small` (2 vcpu, 2GB ram), and `t2.small` (1 vcpu, 2GB ram) and it didn't timeout on any of them. What kind of machine are you using?

We could try making the timeout values configurable via env vars. Right now it's 30s. But ideally we'd understand why it takes so long first.

**Reply by timscha** [reply]
Server is a VM on a KVM platform. 8 GB Ram, 8 Core CPU (and there is enough ressources left). Maybe I can increase the timeout value somehow before you changing everything so we can see if it fixes the problem.

**Reply by ruslandoga** [reply]
You can change [this line](https://github.com/plausible/analytics/blob/ae4ae5d0a0b49ee0fbd5146cdd169455c6924f3d/lib/plausible/application.ex#L34) to be something like `Location.load_all(:timer.seconds(60))` to make the timeout value be 60 seconds. And then build a container image.

```console
$ git clone https://github.com/plausible/analytics
$ cd analytics
$ git checkout v1.5.0-rc.0

$ # my patch: https://gist.github.com/ruslandoga/a0d78a8ca7625abbe7adcc99945af31b
$ curl https://gist.githubusercontent.com/ruslandoga/a0d78a8ca7625abbe7adcc99945af31b/raw/b4f2db9b01880984d45d40fe717e1e9cadf0d795/60s-timeout.diff | git apply

$ git status
HEAD detached at v1.5.0-rc.0
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   lib/plausible/application.ex

no changes added to commit (use "git add" and/or "git commit -a")

$ git diff
diff --git a/lib/plausible/application.ex b/lib/plausible/application.ex
index c0a8b9da..854a2006 100644
--- a/lib/plausible/application.ex
+++ b/lib/plausible/application.ex
@@ -31,7 +31,7 @@ defmodule Plausible.Application do
     OpentelemetryPhoenix.setup()
     OpentelemetryEcto.setup([:plausible, :repo])
     OpentelemetryEcto.setup([:plausible, :clickhouse_repo])
-    Location.load_all()
+    Location.load_all(:timer.seconds(60))
     Supervisor.start_link(children, opts)
   end

$ docker build . -t <username>/analytics:60s-timeout
$ docker push <username>/analytics:60s-timeout
```

**Reply by timscha** [reply]
Thanks @ruslandoga  - I will try that. At the moment I have some problems pushing the image to docker hub.

**Reply by evrpress** [reply]
Thanks @ruslandoga for the detailed explanation on the update. I followed exactly but still get no Regions/Cities tracked. I've checked everything again and all files in place and settings are defined. 
<img width="250" alt="image" src="https://user-images.githubusercontent.com/881729/195976814-ea572fb4-72ce-48af-9dd2-e6a385afdd70.png">
<img width="250" alt="image" src="https://user-images.githubusercontent.com/881729/195976822-26de2a9c-68e8-4add-8dcc-fd5ef56dae83.png">

Is there a place I can check for errors or more info on the problem?

**Reply by ruslandoga** [reply]
👋 @evrpress

> Is there a place I can check for errors or more info on the problem?

You can connect to a running plausible container and check that you get [all the required info](https://github.com/plausible/analytics/blob/aa97e9cf894bcd40bcc073775b599093683599b8/lib/plausible/ingestion/event.ex#L208-L243) from dbip.

```console
$ cd hosting
$ docker compose exec -ti plausible bin/plausible remote
```
```elixir
iex> some_ip_address = "2001:fb1:45:8ff1:61a9:8b71:d13b:f6ff"
iex> Geolix.lookup(some_ip_address, where: :geolocation)
%{
  city: %{names: %{en: "Vadhana"}},
  # >snip<
  subdivisions: [%{names: %{en: "Bangkok"}}]
}
```

In my example above the required info to display cities and regions is not available. Both `city.geoname_id` and `subdivisions[*].iso_code` are missing.

**Reply by ruslandoga** [reply]
Maybe `dbip-city-lite` doesn't include that info, I don't know. You can use [maxmind geolite2](https://dev.maxmind.com/geoip/geolite2-free-geolocation-data?lang=en) instead, though:

```elixir
#  the same steps, but with maxmind's geolite2 city
iex> some_ip_address = "2001:fb1:45:8ff1:61a9:8b71:d13b:f6ff"
iex> Geolix.lookup(some_ip_address, where: :geolocation)
%{
  city: %{
    # has geoname_id, that's used to store the city
    # https://www.geonames.org/1609350
    geoname_id: 1609350, 
    names: %{
      en: "Bangkok",
      # >snip<
    }
  },
  # >snip<
  subdivisions: [
    # has subdivision iso_code, that's used to store region info
    # region is `TH-10`, more info on https://en.wikipedia.org/wiki/ISO_3166-2:TH
    %{geoname_id: 1609348, iso_code: "10", names: %{en: "Bangkok"}}
  ]
}
```

___

### MaxMind GeoLite2 City setup

```console
$ cd hosting
$ export MAXMIND_LICENSE_KEY=LNpsJCCKPis6XvBP # this key is not valid anymore
$ curl -L "https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City&license_key=${MAXMIND_LICENSE_KEY}&suffix=tar.gz" -o geolite2-city.mmdb.gz
$ gunzip geolite2-city.mmdb.gz
```


`docker-compose.yml`
```diff
  plausible:
    image: plausible/analytics:1.5.0-rc.0
    volumes:
      - ./geonames.csv:/etc/app/geonames.csv:ro
-     - ./dbip-city-lite-2022-10.mmdb:/etc/app/dbip-city-lite-2022-10.mmdb:ro
+     - ./geolite2-city.mmdb:/etc/app/geolite2-city.mmdb:ro
```

`plausible-conf.env`
```diff
ADMIN_USER_EMAIL=replace-me
ADMIN_USER_NAME=replace-me
ADMIN_USER_PWD=replace-me
BASE_URL=replace-me
SECRET_KEY_BASE=replace-me
GEONAMES_SOURCE_FILE=/etc/app/geonames.csv
- IP_GEOLOCATION_DB=/etc/app/dbip-city-lite-2022-10.mmdb
+ IP_GEOLOCATION_DB=/etc/app/geolite2-city.mmdb
```

**Reply by evrpress** [reply]
Thanks for the update @ruslandoga !

I've checked my own IP with `Geolix.lookup` and it returned the City in the output, still no luck on the tracking though. So I switched to Maxmind as suggested and the City is there 🎉

If you need more info on the problem I'm happy to help you checking error files etc.

Thanks again for your quick help.

**Reply by anudeepND** [reply]
@ruslandoga do we need to have a cronjob to keep geolite2-city.mmdb up to date?

**Reply by ruslandoga** [reply]
@anudeepND 👋

Yes, and you'd also need to restart the `plausible` container. [Geolix](https://github.com/elixir-geolix/geolix) loads the `mmdb` files in memory and doesn't seem to be watching them for changes. I'll ask Uku if it makes sense to make the updates automatic.

**Reply by ruslandoga** [reply]
PR that switches the geolocation library for the one that automatically loads and updates maxmind dbs: https://github.com/plausible/analytics/pull/2362

**Reply by joshavant** [reply]
@ruslandoga Re: City level geolocation requiring an extra ~1GB of RAM - Would it be possible to run this release on a server with only 1GB of RAM total, for a low traffic website (< 100 pageviews/day)? Would any performance hit only be tied to slow servers and restarts?

**Reply by ruslandoga** [reply]
@joshavant I think it should be possible if you use the built-in geonames file (= no `GEONAMES_SOURCE_FILE` set) and a maxmind geolite2 city mmdb.

**Reply by joshavant** [reply]
@ruslandoga Sorry if I'm missing something but... there is a built-in genomes file?

What's the advantages/differences of using the CSV file in the comments above (e.g. https://s3.eu-central-1.wasabisys.com/plausible-application/geonames.csv )?

**Reply by ruslandoga** [reply]
@joshavant it's the one that's included by default in the `:location` dependency, `cities500.txt`: https://github.com/plausible/location#citie. And it's used when `GEONAMES_SOURCE_FILE` is not set.

> What's the advantages/differences of using the CSV file in the comments above 

It has more geoname id -> city info mappings, that's all.

**Comment by rphlo** [comment]
For my second point which was hidding the "Regions" and "Cities" Pills in the locations widget if this feature is not enabled, I implemented myself.
Commit is here https://github.com/rphlo/analytics/commit/6a6495ec2c201879e8b013a12ea3b8ea75df8d40 and could be added to the rc1 version...

**Reply by ruslandoga** [reply]
Nice! We could do it automatically though, without a flag, by checking if the ip db has city info data.

**Reply by rphlo** [reply]
Sure, that be nice, but that is beyond my know how with elixir

**Comment by mannp** [comment]
Done the postgres 12 to 14 upgrade successfully and was already on the specified clickhouse version.

Seems clickhouse is not happy with the migration....any pointers as to how to solve it please? Thanks in advance :)


```
Loading plausible..
Starting dependencies..
Starting repos..
Running migrations for Elixir.Plausible.Repo
Running migrations for Elixir.Plausible.ClickhouseRepo
** (Clickhousex.Error) Code: 10. DB::Exception: Wrong column name. Cannot find column `entry.meta` to drop. (NOT_FOUND_COLUMN_IN_BLOCK) (version 22.6.9.11 (official build))

    (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:932: Ecto.Adapters.SQL.raise_sql_call_error/1
    (elixir 1.13.4) lib/enum.ex:1593: Enum."-map/2-lists^map/1-0-"/2
    (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:1024: Ecto.Adapters.SQL.execute_ddl/4
    (ecto_sql 3.8.3) lib/ecto/migration/runner.ex:352: Ecto.Migration.Runner.log_and_execute_ddl/3
    (ecto_sql 3.8.3) lib/ecto/migration/runner.ex:117: anonymous fn/6 in Ecto.Migration.Runner.flush/0
    (elixir 1.13.4) lib/enum.ex:2396: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ecto_sql 3.8.3) lib/ecto/migration/runner.ex:116: Ecto.Migration.Runner.flush/0
    (stdlib 3.17.1) timer.erl:166: :timer.tc/1
bash: started: command not found
```

**Reply by ruslandoga** [reply]
👋 @mannp

That's a strange one. Somehow `entry.meta` column is missing in your sessions table. I guess a quick and dirty way would be to override the migration file with a mounted volume. If anyone else has that problem, we'd need to look more into it.

---

Please create a file called `fix` (name doesn't matter) somewhere locally:

```elixir
# based on https://github.com/plausible/analytics/blob/master/priv/clickhouse_repo/migrations/20220421161259_remove_entry_props.exs
defmodule Plausible.ClickhouseRepo.Migrations.RemoveEntryProps do
  use Ecto.Migration

  def change do
    alter table(:sessions) do
      remove_if_exists :"entry.meta"
    end
  end
end
```

and then mount it into the `plausible` container to overwrite the existing migration

```diff
  plausible:
    image: plausible/analytics:1.5.0-rc.0
+   volumes:
+     - ./fix:/app/lib/plausible-0.0.1/priv/clickhouse_repo/migrations/20220421161259_remove_entry_props.exs
```

**Reply by ruslandoga** [reply]
It would need to be done once. After the migration runs successfully, it won't be run again, so removing the mounted file would be fine.

**Reply by ukutaht** [reply]
@ruslandoga Thanks, if there are more issues with this then it wouldn't hurt to make this change in the main codebase as well.

**Reply by mannp** [reply]
Thanks @ruslandoga, gave it a try and now have this error;

```
Running migrations for Elixir.Plausible.ClickhouseRepo
** (CompileError) lib/plausible-0.0.1/priv/clickhouse_repo/migrations/20220421161259_remove_entry_props.exs:7: undefined function remove_if_exists/1 (expected Plausible.ClickhouseRepo.Migrations.RemoveEntryProps to define such a function or for it to be imported, but none are available)
    (elixir 1.13.4) lib/code.ex:404: Code.validated_eval_string/3
```

**Reply by ruslandoga** [reply]
@mannp sorry, the migration I wrote is indeed incorrect. How about just doing nothing in it? That would definitely work with the same end result.

```elixir
defmodule Plausible.ClickhouseRepo.Migrations.RemoveEntryProps do
  use Ecto.Migration

  def change do
  end
end
```

**Reply by mannp** [reply]
> @mannp sorry, the migration I wrote is indeed incorrect. How about just doing nothing in it? That would definitely work with the same end result.
> 
> ```elixir
> defmodule Plausible.ClickhouseRepo.Migrations.RemoveEntryProps do
>   use Ecto.Migration
> 
>   def change do
>   end
> end
> ```

Thanks, that worked.

**Comment by mannp** [comment]
Hi @ruslandoga has GEOLITE2_COUNTRY_DB now been depreciated? as I get GeoLite2-City.mmdb updated automatically.

**Reply by ruslandoga** [reply]
No, it's not deprecated and would continue to [work as before:](https://github.com/plausible/analytics/blob/ae4ae5d0a0b49ee0fbd5146cdd169455c6924f3d/config/runtime.exs#L113-L120) if `IP_GEOLOCATION_DB` is not set, fallback to `GEOLITE2_COUNTRY_DB`, and if `GEOLITE2_COUNTRY_DB` is not set, fallback to `dbip-country-lite` [downloaded](https://github.com/plausible/analytics/blob/master/lib/mix/tasks/download_country_database.ex) when container image [was built.](https://github.com/plausible/analytics/blob/ae4ae5d0a0b49ee0fbd5146cdd169455c6924f3d/Dockerfile#L41)

At least right now, `IP_GEOLOCATION_DB` and `GEOLITE2_COUNTRY_DB` are equivalent, but when both are specified, `IP_GEOLOCATION_DB` takes precedence.

**Reply by mannp** [reply]
> GEOLITE2_COUNTRY_DB

Great, thanks for the clarification.

**Comment by paskal** [comment]
Nitpick, previous versions in dockerhub were named like `v1.4.4`, I propose to release future versions with leading `v` also, and add such tag for already published image.

**Reply by ruslandoga** [reply]
I agree. For some reason https://github.com/docker/metadata-action drops the `v`: https://github.com/docker/metadata-action#semver

@ukutaht should we maybe keep the `v` in the container image tag?

**Reply by ukutaht** [reply]
Yes we should for consistency with older versions

**Reply by ruslandoga** [reply]
https://github.com/plausible/analytics/pull/2340

**Comment by ruslandoga** [comment]
`v1.5.0-rc.1` discussion: https://github.com/plausible/analytics/discussions/2460

**Comment by ruslandoga** [comment]
`v2.0.0-rc.0` discussion: https://github.com/plausible/analytics/discussions/2896
